code/components/openmetrics/openmetrics.cpp

```diff
@@ -0,0 +1,44 @@
+#include "openmetrics.h"
+
+using namespace std;
+
+/**
+ * create a singe metric from the given input
+ **/
+string createMetric(const string &metricName, const string &help, const string &type, const string &value)
+{
+    return "# HELP " + metricName + " " + help + "\n# TYPE " + metricName + " " + type + "\n" + metricName + " " + value + "\n";
+}
+
+/**
+ * Generate the MetricFamily from all available sequences
+ * @returns the string containing the text wire format of the MetricFamily
+ **/
+string createSequenceMetrics(string prefix, std::vector<NumberPost *> *numbers)
+{
+    string res = "";
+
+    for (int i = 0; i < (*numbers).size(); ++i)
+    {
+        auto number = (*numbers)[i];
+        // only valid data is reported (https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#missing-data)
+        if (number->ReturnValue.length() > 0)
+        {
+            auto label = number->name;
+
+            // except newline, double quote, and backslash (https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#abnf)
+            // to keep it simple, these characters are just remove from the label
+            label = ReplaceString(label, "\\", "");
```

Comment:
If you make `label` a `string` the in-place `replaceString(std::string& s, std::string const& toReplace, std::string const& replaceWith)` can be used, saving a bit of memory.

Reply:
This will not work, because replaceString() only replaces the 1st occurrence of `toReplace`. Maybe I will implement another function to do the replacing for all occurrence.